Core/Utilities/MySQL.py

```diff
@@ -849,276 +809,273 @@ def _createTables( self, tableDict, force = False ):
+        if 'PrimaryKey' in thisTable:
+          if isinstance(thisTable['PrimaryKey'], basestring):
+            cmdList.append('PRIMARY KEY ( `%s` )' % thisTable['PrimaryKey'])
           else:
-            cmdList.append( 'PRIMARY KEY ( %s )' % ", ".join( [ "`%s`" % str( f ) for f in thisTable['PrimaryKey'] ] ) )
+            cmdList.append('PRIMARY KEY ( %s )' % ", ".join(["`%s`" % str(f) for f in thisTable['PrimaryKey']]))
 
-        if thisTable.has_key( 'Indexes' ):
+        if 'Indexes' in thisTable:
           indexDict = thisTable['Indexes']
           for index in indexDict:
-            indexedFields = '`, `'.join( indexDict[index] )
-            cmdList.append( 'INDEX `%s` ( `%s` )' % ( index, indexedFields ) )
+            indexedFields = '`, `'.join(indexDict[index])
+            cmdList.append('INDEX `%s` ( `%s` )' % (index, indexedFields))
 
-        if thisTable.has_key( 'UniqueIndexes' ):
+        if 'UniqueIndexes' in thisTable:
           indexDict = thisTable['UniqueIndexes']
           for index in indexDict:
-            indexedFields = '`, `'.join( indexDict[index] )
-            cmdList.append( 'UNIQUE INDEX `%s` ( `%s` )' % ( index, indexedFields ) )
+            indexedFields = '`, `'.join(indexDict[index])
+            cmdList.append('UNIQUE INDEX `%s` ( `%s` )' % (index, indexedFields))
         if 'ForeignKeys' in thisTable:
           thisKeys = thisTable['ForeignKeys']
           for key, auxTable in thisKeys.iteritems():
 
-            forTable = auxTable.split( '.' )[0]
+            forTable = auxTable.split('.')[0]
             forKey = key
             if forTable != auxTable:
-              forKey = auxTable.split( '.' )[1]
+              forKey = auxTable.split('.')[1]
 
             # cmdList.append( '`%s` %s' % ( forTable, tableDict[forTable]['Fields'][forKey] )
-            cmdList.append( 'FOREIGN KEY ( `%s` ) REFERENCES `%s` ( `%s` )'
-                            ' ON DELETE RESTRICT' % ( key, forTable, forKey ) )
+            cmdList.append('FOREIGN KEY ( `%s` ) REFERENCES `%s` ( `%s` )'
+                           ' ON DELETE RESTRICT' % (key, forTable, forKey))
 
-        if thisTable.has_key( 'Engine' ):
+        if 'Engine' in thisTable:
           engine = thisTable['Engine']
         else:
           engine = 'InnoDB'
 
-        if thisTable.has_key( 'Charset' ):
+        if 'Charset' in thisTable:
           charset = thisTable['Charset']
         else:
           charset = 'latin1'
 
-        cmd = 'CREATE TABLE `%s` (\n%s\n) ENGINE=%s DEFAULT CHARSET=%s' % ( table, ',\n'.join( cmdList ), engine, charset )
-        retDict = self._update( cmd, debug = True )
+        cmd = 'CREATE TABLE `%s` (\n%s\n) ENGINE=%s DEFAULT CHARSET=%s' % (table, ',\n'.join(cmdList), engine, charset)
+        retDict = self._update(cmd)
         if not retDict['OK']:
           return retDict
-        self.log.info( 'Table %s created' % table )
+        self.log.debug('Table %s created' % table)
```

Comment:
This can be left at Info. This is only printed at the first service invocation when no tables are there yet

Reply:
It could be, but does it make sense ? the service anyway prints it if creates table.

Reply:
If so, then this one can be dropped completely

Reply:
OK. If you don't mind, I will do it in the next PR for log cleaning, such that these changes are in the next patch release and can already go in prod next week